backend/src/cusco/storage.py

```
@@ -0,0 +1,350 @@
+        """Atomic replace: wipe + reinsert + update meta. Returns inserted count.
+
+        The entire operation runs inside an explicit transaction; on any
+        error we `ROLLBACK` so a partial write never leaves the table in
+        an inconsistent state.
+        """
+        count = 0
+        with closing(get_connection()) as conn:
+            self._ensure_schema(conn)
+            conn.execute("BEGIN")
+            try:
+                conn.execute(f'DELETE FROM "{self.name}"')
```

Ruff (0.15.10)

[error] Possible SQL injection vector through string-based query construction

(S608)